src/libs/SearchParser/searchParser.peggy

```diff
@@ -5,147 +5,162 @@
-    }
-
-    return withDefaults;
-  }
+query = _ filters:filterList? _ { return applyDefaults(filters); }
 
 filterList
   = head:filter tail:(logicalAnd filter)* {
-      const allFilters = [head, ...tail.map(([_, filter]) => filter)].filter(filter => filter !== null);
+      const allFilters = [head, ...tail.map(([_, filter]) => filter)]
+        .filter((filter) => filter !== null)
+        .filter(Boolean);
       if (!allFilters.length) {
-      	return null;
+        return null;
+      }
+
+      const keywords = allFilters.filter(
+        (filter) =>
+          filter.left === "keyword" || filter.right?.left === "keyword"
+      );
+      const nonKeywords = allFilters.filter(
+        (filter) =>
+          filter.left !== "keyword" && filter.right?.left !== "keyword"
+      );
+
+      if (!nonKeywords.length) {
+        return keywords.reduce((result, filter) =>
+          buildFilter("or", result, filter)
+        );
       }
-     const keywords = allFilters.filter((filter) => filter.left === "keyword" || filter.right?.left === "keyword")
-     const nonKeywords = allFilters.filter((filter) => filter.left !== "keyword" && filter.right?.left !== "keyword")
-     if(!nonKeywords.length){
-     	return keywords.reduce((result, filter) => buildFilter("or", result, filter))
-     }
-     if(!keywords.length){
-     	return nonKeywords.reduce((result, filter) => buildFilter("and", result, filter))
-     }
-
-     return buildFilter("and", keywords.reduce((result, filter) => buildFilter("or", result, filter)), nonKeywords.reduce((result, filter) => buildFilter("and", result, filter)))
-
-      return allFilters.reduce((result, filter) => buildFilter("and", result, filter));
+      if (!keywords.length) {
+        return nonKeywords.reduce((result, filter) =>
+          buildFilter("and", result, filter)
+        );
+      }
+
+      return buildFilter(
+        "and",
+        keywords.reduce((result, filter) => buildFilter("or", result, filter)),
+        nonKeywords.reduce((result, filter) =>
+          buildFilter("and", result, filter)
+        )
+      );
     }
 
 filter
-  = _ field:key? _ op:operator? _ value:identifier {
-      if (isDefaultField(field)) {
-        updateDefaultValues(field, value.trim());
-        return null;
-      }
+  = @(
+      policyIDFilter
+      / standardFilter
+      / defaultFilter
+      / multiSelectFilter
+      / freeTextFilter
+    )
+
+defaultFilter
+  = _ key:defaultKey _ op:operator _ value:identifier {
+      updateDefaultValues(key, value.trim());
+    }
 
-      if (isPolicyID(field)) {
-        updateDefaultValues(field, value.trim());
-        return null;
-      }
+freeTextFilter
+  = _ value:identifier _ { return buildFilter("eq", "keyword", value.trim()); }
 
-      if (!field && !op) {
-        return buildFilter('eq', 'keyword', value.trim());
-      }
+multiSelectFilter
+  = _ field:multiselectKey _ op:operator _ value:identifier {
+      return buildFilter(op, field, value);
```

Comment:
Thinking more about this, it might be better to send multiSelect values as a json array, instead of a comma separated string. I realized that it's possible for values to contain commas in a quoted string, in which case it'd be challenging to parse this in the backend. Example:

```
op: 'eq',
left: 'category',
right: ['Hotel', 'Travel', 'Electricity, water and gas'],
```